ProcessMaker/processmaker
# Fix trigger boundary events with multi-instance subprocess

## Issue & Reproduction Steps
Fix trigger boundary events with multi-instance subprocess.

## Solution
- Fix Job that catch signal.
- Fix missing signal_events property in process_request table when a boundary event is activated.

## How to Test
- Create A Collection
- Add records to the collection
- Enable the Collection events
- Import or create a process similar to the attached one
- Assign to the boundary signal event to one of the events of the collection
- Start the process
- Trigger one of the collection events
- The CallActivity should be closed and continue to the task that follows the boundary event

https://user-images.githubusercontent.com/8028650/144326886-c72baa1b-121a-46a9-b8d4-3864a222080e.mp4

Main Process:
[Signals.json.txt](https://github.com/ProcessMaker/processmaker/files/7637564/Signals.json.txt)
Sub Process:
[subprocess.json.txt](https://github.com/ProcessMaker/processmaker/files/7637565/subprocess.json.txt)
Collection:
[persons.json.txt](https://github.com/ProcessMaker/processmaker/files/7637572/persons.json.txt)

## Related Tickets & Packages
- [FOUR-4407](https://processmaker.atlassian.net/browse/FOUR-4407)

Requires: https://github.com/ProcessMaker/nayra/pull/212
Requires: https://github.com/ProcessMaker/package-collections/pull/293

## Limitation
The case in which a multi-instance parallel subprocess that triggers immediately multiple times the same signal is not yet working as expected in PM4 Core, seems it will need some additional work in the Extended CallActivity class.

This Escenario is working in Nayra stand-alone.
Main process:
![image](https://user-images.githubusercontent.com/8028650/145644987-8225c133-01b6-40e7-ab89-a47b42225570.png)
Sub process:
![image](https://user-images.githubusercontent.com/8028650/145645010-249380d2-5b75-4231-88c1-4d038d49f394.png)


## Code Review Checklist
